cmake/LLVM.cmake

```diff
@@ -91,12 +91,6 @@ function(fetch_or_build_slang_llvm)
             target_compile_options(slang-llvm PRIVATE -wd4244 /Zc:preprocessor)
         endif()
 
-        if(NOT LLVM_ENABLE_RTTI)
```

Comment:
I think we should keep this `if(NOT LLVM_ENABLE_RTTI)` block here for the `USE_SYSTEM_LLVM` option to work. Some distributions may have LLVM without RTTI, and removing this would cause build errors with that option on such distributions.

I think this block shouldn't be running on builds with `build-llvm.sh` because of the flag that was added in this PR.